posthog/permissions.py

```diff
@@ -156,6 +136,10 @@ def has_permission(self, request, view) -> bool:
         try:
             view.team  # noqa: B018
         except Team.DoesNotExist:
+            # Legacy compatibility for old endpoints without the team in the path
+            if view.legacy_team_compatibility:
+                return request.user.is_authenticated and request.user.team is not None
```

Comment:
This shouldn't be needed, since `view.team` already returns `request.user.team` for `legacy_team_compatibility` viewsets and fails at `assert team is not None` if the user has no current project set

Reply:
Ah okay cool